lessons/textstats.md

```diff
@@ -0,0 +1,571 @@
+
+    t_length_max + 100 - (t_length_max % 100)
+    
+    t_length_max = 1026 + 100 = 1126
+    t_length_max / 100 = 10 with a remainder of 26
+    1126 - 26 = 1100 ==> y_max
+
+We assign the value of this bit of math to `y_max`, knowing that in the future we won't ever have to worry about this, unless our texts decrease or increase radically in size, in which case we can just change 100 to 10 or to 1000 and we are good to go.
+
+One final bit of fitting the graph to our data requires that we tell `matplotlib` where to place the bars: we need one spot on the x-axis for each bar. `numpy`'s range function does this for us, and, like the `x_max` above, I add one so that our first bar starts further in on the graph. (Try removing the `+ 1` from that line: the whole bar chart shifts to the left.)
+
+Here's what all of that looks like in our script:
+
+
+    x_max = len(eachLine) + 1
+    t_length_max = sorted(t_length, reverse=True)[0] 
+    y_max = t_length_max + 100 - (t_length_max % 100)
+    x_loc = np.arange(len(eachLine)) + 1
+
+What follows, to my mind, occasionally border on voodoo, but there's not much more to do but to follow the advice and practice of others. (I have stared at dozens of bar chart scripts in an attempt to grok the logic of `pyplot` but it mostly continues to elude me.) Since we want to show two bars simultaneously, we are actually going to create two plots, imaginatively labeled here `plot_len` and `plot_lex` -- they were `plot_1` and `plot_2` for a long time but that seemed too Seussian to be useful. Inside the parentheses, the script tells `pyplot` where to locate the bars on the x-axis, how tall they should be, how wide, where their bottoms should be located -- we are not stacking bars here, so both start at 0, and that one bar is to be red and the other yellow, and they both should be centered.
+
+The next five lines dictate the dimensions of our graph, label the two axes, give the graph a title, and label each of the bars. Once all that is done, we tell `pyplot` to show the results, or, as you'll see in the completed script at the end of this section, you can save the graph to a file.
+
+
+    plot_len = plt.bar(x_loc, t_length, width=0.75, bottom=0, color='r', align='center')
+    plot_lex = plt.bar(x_loc, t_lexicon, width=0.75, bottom=0, color='y', align='center')
+    
+    plt.axis([0, x_max, 0, y_max])
+    plt.xlabel('Texts')
+    plt.ylabel('Word Count')
+    plt.title('Length and Lexicon')
+    plt.xticks(x_loc, t_name, rotation='vertical') 
+    
+    plt.show()
+
+
+![png](textstats_files/textstats_43_0.png)
```

Comment:
@johnlaudun Currently your images aren't showing up in the live preview. Can you change these image lines to use the Programming Historian [guidelines for figures](http://programminghistorian.org/new-lesson-workflow#figures)? So in this case, the line would become something like:

    {% include figure.html src="../images/textstats_43_0.png" caption="Results of pyplot (or some other caption)" %}